analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -64,4 +64,17 @@ public static CollectionConstraint ObjectCreationConstraint(ProgramState state,
         operation.DimensionSizes.Any(x => x.ConstantValue.Value is 0)
             ? CollectionConstraint.Empty
             : CollectionConstraint.NotEmpty;
+
+    public static ProgramState ApplyConstraints(ProgramState state, IPropertyReferenceOperationWrapper operation, ISymbol instanceSymbol)
```

Comment:
name is a bit generic, but the type of `operation` restricts the usage.
If you have any suggestions, please let me know.

Reply:
Maybe `LearnFrom`?

Reply:
I like it way more, thanks!